From e1f29013dd6eeb77db7fb0ad6e9343cd880ccbde Mon Sep 17 00:00:00 2001
From: Michael Schroeder <mls@suse.de>
Date: Thu, 3 Mar 2016 15:18:34 +0100
Subject: [PATCH 8/9] Make _Pool_tmpspace definition internal

---
 src/pool.h | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/src/pool.h b/src/pool.h
index 4a2089d..b9e2ed6 100644
--- a/src/pool.h
+++ b/src/pool.h
@@ -34,9 +34,6 @@ extern "C" {
 #define SYSTEMSOLVABLE		1
 
 
-/* how many strings to maintain (round robin) */
-#define POOL_TMPSPACEBUF 16
-
 /*----------------------------------------------- */
 
 struct _Repo;
@@ -52,12 +49,20 @@ typedef struct _Datapos {
   Id dp;
 } Datapos;
 
+
+#ifdef LIBSOLV_INTERNAL
+
+/* how many strings to maintain (round robin) */
+#define POOL_TMPSPACEBUF 16
+
 struct _Pool_tmpspace {
   char *buf[POOL_TMPSPACEBUF];
   int   len[POOL_TMPSPACEBUF];
   int   n;
 };
 
+#endif
+
 struct _Pool {
   void *appdata;		/* application private pointer */
 
-- 
2.5.0

